worlds/stardew_valley/logic/pet_logic.py

```diff
@@ -20,21 +18,25 @@ def __init__(self, *args, **kwargs):
 
 
 class PetLogic(BaseLogic[Union[RegionLogicMixin, ReceivedLogicMixin, TimeLogicMixin, ToolLogicMixin]]):
-    def has_hearts(self, hearts: int = 1) -> StardewRule:
-        if hearts <= 0:
+    def has_pet_hearts(self, hearts: int = 1) -> StardewRule:
+        assert hearts >= 0, "You can't have negative hearts with a pet."
+        if hearts == 0:
             return True_()
-        if self.options.friendsanity == Friendsanity.option_none or self.options.friendsanity == Friendsanity.option_bachelors:
-            return self.can_befriend_pet(hearts)
-        return self.received_hearts(NPC.pet, hearts)
 
-    def received_hearts(self, npc: Union[str, Villager], hearts: int) -> StardewRule:
-        if isinstance(npc, Villager):
-            return self.received_hearts(npc.name, hearts)
-        return self.logic.received(self.heart(npc), math.ceil(hearts / self.options.friendsanity_heart_size))
+        if self.content.features.friendsanity.is_enabled:

```

Comment:
Is that correct? Pets aren't shuffled on lower friendsanity settings

Reply:
I think you are right in that this is not right. Looking at items.py, it seems I broke that. I'll fix it when I have a moment.

Reply:
Fixed